src/meta/meta_turtle.c

```diff
@@ -226,6 +258,13 @@ __wt_turtle_init(WT_SESSION_IMPL *session)
         if (ret != 0) {
             WT_RET(__wt_remove_if_exists(session, WT_METADATA_TURTLE, false));
             loadTurtle = true;
+        } else {
+            /*
+             * Check the version in the turtle file to validate whether we can start up on the
+             * turtle file version seen. Return an error if we can't. Only check if we either didn't
+             * run salvage or if salvage didn't fail to read it.
+             */
+            WT_RET(__turtle_validate_version(session));
```

Comment:
I think this needs to move after the backup tests just below this. Maybe here set a `validate` boolean, and then call this function later.